test/array.cpp

```diff
@@ -582,3 +582,69 @@ TEST(Array, InitializerListAndDim4) {
 
     ASSERT_ARRAYS_EQ(A, B);
 }
+
+TEST(Broadcast, Simple1) {
```

Comment:
Can you create parameterized tests for the input dims? You also want to compare this functionality to the tiled version of the same operation. Additionally you should make sure that the kernels generated and the operations performed by these functions are what you expect.